tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/checkedexpr/CheckedExprNegativeTest.java

```diff
@@ -35,15 +35,14 @@
     public void testSemanticErrors() {
         CompileResult compile = BCompileUtil.compile(
                 "test-src/expressions/checkedexpr/checked_expr_semantics_negative.bal");
-        Assert.assertEquals(compile.getErrorCount(), 4, compile.toString());
+        Assert.assertEquals(compile.getErrorCount(), 5, compile.toString());
         BAssertUtil.validateError(compile, 0, "invalid usage of the 'check' expression " +
                 "operator: no expression type is equivalent to error type", 11, 25);
-        BAssertUtil.validateError(compile, 1, "invalid usage of the 'check' expression " +
-                "operator: all expression types are equivalent to error type", 16, 25);
-        BAssertUtil.validateError(compile, 2, "invalid usage of the 'check' expression " +
-                "operator: all expression types are equivalent to error type", 30, 25);
+        BAssertUtil.validateError(compile, 1, "incompatible types: expected 'string', found 'never'", 16, 19);
```

Comment:
Example is here is 
```
function testCheckedExprSemanticErrors2() returns error? {
    string line = checkpanic readLineError();
}
```

The diagnostic is correct, But I think we can improve this.

Reply:
@hasithaa Shall I fix this in a separate PR?

Reply:
yes. Please create an issue, we can fix this later.

Reply:
Created.(https://github.com/ballerina-platform/ballerina-lang/issues/29696)